pkg/cmd/cli/describe/chaindescriber.go

```diff
@@ -0,0 +1,267 @@
+package describe
+
+import (
+	"fmt"
+	"strings"
+
+	utilerrors "github.com/GoogleCloudPlatform/kubernetes/pkg/util/errors"
+	dot "github.com/awalterschulze/gographviz"
+	"github.com/golang/glog"
+	"github.com/gonum/graph"
+
+	osgraph "github.com/openshift/origin/pkg/api/graph"
+	buildapi "github.com/openshift/origin/pkg/build/api"
+	buildedges "github.com/openshift/origin/pkg/build/graph"
+	buildutil "github.com/openshift/origin/pkg/build/util"
+	"github.com/openshift/origin/pkg/client"
+	imageapi "github.com/openshift/origin/pkg/image/api"
+	imageedges "github.com/openshift/origin/pkg/image/graph"
+	imagegraph "github.com/openshift/origin/pkg/image/graph/nodes"
+	"github.com/openshift/origin/pkg/util/parallel"
+)
+
+// ChainDescriber generates extended information about a chain of
+// dependencies of an image stream
+type ChainDescriber struct {
+	client.Interface
+	Output string
+}
+
+// NewChainDescriber returns a new ChainDescriber
+func NewChainDescriber(c client.Interface, out string) *ChainDescriber {
+	return &ChainDescriber{Interface: c, Output: out}
+}
+
+// MakeGraph will create the graph of all build configurations and the image streams
+// they point to via image change triggers in the provided namespace
+func (d *ChainDescriber) MakeGraph(namespace string) (osgraph.Graph, error) {
+	g := osgraph.New()
+
+	loaders := []GraphLoader{
+		&bcLoader{namespace: namespace, lister: d.Interface},
+		&isFromBuildConfigLoader{namespace: namespace, lister: d.Interface, bc: &bcLoader{namespace: namespace, lister: d.Interface}},
+	}
+	loadingFuncs := []func() error{}
+	for _, loader := range loaders {
+		loadingFuncs = append(loadingFuncs, loader.Load)
+	}
+
+	if errs := parallel.Run(loadingFuncs...); len(errs) > 0 {
+		return g, utilerrors.NewAggregate(errs)
+	}
+
+	for _, loader := range loaders {
+		loader.AddToGraph(g)
+	}
+
+	buildedges.AddAllInputOutputEdges(g)
+	imageedges.AddAllImageStreamRefEdges(g)
```

Comment:
Aaah, thinking about it BuildOutput and BuildInputImage edges are enough. This means though that we cut off ImageStream nodes. I can try that and see about any consequences as we go on